src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -694,6 +702,16 @@ default int numCigarElements(){
      */
     void setAttribute( final String attributeName, final byte[] attributeValue );
 
+    /**
+     * This is used to access the transient attribute store in the underlying data type. This is used to store temporary
+     * attributes and cached data that will not be serialized or written out as a record.
+     *
+     * NOTE: This is an advanced use case for GATKRead and you should probably use setAttribute() instead
+     * @param key key under which the value will be stored
+     * @param value value to be keyed upon
+     */
+    public void setTransientAttribute(final Object key, final Object value);
+
```

Comment:
There really should be a `clearTransientAttribute` method as well (SAMRecord has one so the GATK read adapter should pass that through).

Reply:
This needs a basic set/get/clear test in `GATKReadAdaptersUnitTest`, and a roundtrip test in SAMRecordToGATKReadAdapterSerializerUnitTest to make sure we're really not serializing the attributes.

Reply:
Done. (not I haven't added a clearTransientAttributes() call because it doesn't exist in the underlying SAMRecord code)

Reply:
@jamesemery see SAMRecord.removeTransientAttribute

Reply:
@cmnbroad To clarify, I did not add a clearTransientAttribute_**s**_() method. There is no exposed sam record method for removing ALL the transient attributes at once like there is for normal attributes.